src/server/components/dashboard/controllers/listsExportController.ts

```diff
@@ -0,0 +1,94 @@
+import type { NextFunction, Request, Response } from "express";
+import type { ListItem } from "server/models/types";
+import type { ListItemJsonData } from "server/models/listItem/providers/deserialisers/types";
+
+import { prisma } from "server/models/db/prisma-client";
+import { json2csv } from "json-2-csv";
+import { Readable } from "node:stream";
+import { startCase } from "lodash";
+import { format } from "date-fns";
+import { logger } from "server/services/logger";
+import { AuditEvent } from "@prisma/client";
+
+type ListWithJsonData = Array<ListItem & { jsonData: ListItemJsonData }>;
+
+export async function listsExportController(req: Request, res: Response, next: NextFunction) {
+  try {
+    const [result] = await prisma.$transaction([
+      prisma.list.findUnique({
+        where: {
+          id: Number(req.params.listId),
+        },
+        include: {
+          items: true,
+          country: true,
+        },
+      }),
+      prisma.audit.create({
```

Comment:
Create the audit at the end. There is still actually quite a lot of code being executed between here and when the user actually receives the CSV. If something goes wrong with parsing or conversion, then this audit won't be relevant.

Reply:
Makes sense 👍🏿